### DISCUSSION TITLE
How to Choose Loss Functions

### DISCUSSION THREAD

**Discussion by kkmcgg** [NSCC-COGS/Aestheta#61] [Guides]
https://machinelearningmastery.com/how-to-choose-loss-functions-when-training-deep-learning-neural-networks/

A deep learning model only learns when it knows how wrong it is. How it determines how 'wrong' it is is determined by the loss function. There are several loss functions to choose from! Most generally you would ask things like - am I trying to classify two distinct features or more? am I trying to model a numerical signal? are my samples distributed in a bell curve?  Different questions and data sets require different loss functions to be best modelled. This is a pretty ok guide to learn more!

TLDR: 
If you're doing binary classification (two classes), try 'Binary Cross-Entropy' (aka `loss='binary_crossentropy'` in keras)
If you're classifing more than two classes, try 'Cross-entropy' (aka` loss='categorical_crossentropy' `in keras)
For regression (non-classification) models, try 'Mean Squared Error Loss' (aka` loss='mean_squared_error' `in keras)
For autoencoders and GANs, try 'Kullback Leibler Divergence' (aka `loss='kullback_leibler_divergence'` in keras)
